### ISSUE TITLE
Index errors when using the IN operator with embedded collection types

### ISSUE THREAD

**Issue by obi1kenobi** [orientechnologies/orientdb#6087]
Setup, using OrientDB 2.1.16:
```
CREATE CLASS Foo
CREATE PROPERTY Foo.bar EmbeddedSet String
CREATE INDEX Foo.bar NOTUNIQUE
INSERT INTO Foo SET bar = ['abc', 'def']
INSERT INTO Foo SET bar = ['abcd', 'defg']
INSERT INTO Foo SET bar = ['xyz', '123']
```

[Per the docs](http://orientdb.com/docs/last/SQL-Where.html#conditional-operators), the following two queries should be identical and should hit the index:
```
SELECT * FROM Foo WHERE bar CONTAINS 'xyz'
SELECT * FROM Foo WHERE 'xyz' IN bar
```

The first query hits the index (as evidenced by `EXPLAIN`) and works as expected. However, the second query does not hit the index (`EXPLAIN` says 3 records were read, and `involvedIndexes` is null) and causes the following error to be printed to the logs:

```
Error on using index Foo.bar in query 'SELECT * FROM Foo WHERE 'xyz' IN bar'. Probably you need to rebuild indexes. Now executing query using cluster scan
java.lang.IllegalArgumentException: Key 'xyz' is not valid
	at com.orientechnologies.orient.core.sql.operator.OQueryOperatorIn.executeIndexQuery(OQueryOperatorIn.java:81)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searchForIndexes(OCommandExecutorSQLSelect.java:1929)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searchInClasses(OCommandExecutorSQLSelect.java:922)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.assignTarget(OCommandExecutorSQLResultsetAbstract.java:199)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.assignTarget(OCommandExecutorSQLSelect.java:493)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.executeSearch(OCommandExecutorSQLSelect.java:475)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.execute(OCommandExecutorSQLSelect.java:446)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:90)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLExplain.execute(OCommandExecutorSQLExplain.java:51)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:90)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstractPaginatedStorage.java:1433)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginatedStorage.java:1406)
	at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:67)
	at com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostCommand.execute(OServerCommandPostCommand.java:81)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service(ONetworkProtocolHttpAbstract.java:180)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute(ONetworkProtocolHttpAbstract.java:627)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```

This bug is causing significant performance degradation in my project and making the error logs near-useless.

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

It seems to be a critical one, I'll take a look at that asap

Thanks

Luigi

**Comment by obi1kenobi** [comment]
Hey Luigi,

I see that version 2.1.17 came out -- congrats! Is the fix for bug included, or is it slated for a future release?

Predrag

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

No, unfortunately we could not manage to fix it, it will be in next hotfix

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

I just pushed a fix on 2.1.x and develop, the fix will be available in next 2.1.18 and 2.2 GA

Thanks

Luigi
